core/cap-0063.md

```diff
@@ -0,0 +1,496 @@
+
+If an archived Soroban entry has not yet been evicted and still exists in the live BucketList, it is metered and charged based
+on live Soroban state limits and fees. If an archived Soroban entry has been evicted, it is metered and charged based
+on classic/evicted limits and fees. Additionally, evicted keys must be declared in the `evictedSorobanEntries` bitmap.
+
+##### Failure Codes
+
+If an evicted entry is included in the tx's footprint but is not specified via the `evictedSorobanEntries` bitmap, the
+tx will fail with the `INVOKE_HOST_FUNCTION_ENTRY_ARCHIVED` code. If there are insufficient resources for entry restoration,
+the tx will fail with the `INVOKE_HOST_FUNCTION_RESOURCE_LIMIT_EXCEEDED` code.
+
+##### Meta
+
+Whenever an entry is restored, it will be emitted as a `LedgerEntryChangeType` of type `LEDGER_ENTRY_RESTORE`.
+This will include the complete LedgerEntry of the restored entry. It does not matter if the entry has been
+evicted or not, the meta produced will be the same.
+
+If a restored entry is modified during host function invocation, another `LedgerEntryChangeType` will be
+emitted corresponding to the entry change in addition to the `LEDGER_ENTRY_RESTORE` restore event.
+This is similar to the current meta schema, where the starting value of a LedgerEntry is emitted as
+`LEDGER_ENTRY_STATE` followed by the change. This is similar, except that `LEDGER_ENTRY_STATE` is replaced
+by `LEDGER_ENTRY_RESTORE`.
+
+#### Deprecation of RestoreFootprintOp(?)
+
+With the addition of automatic restoration, there is currently no need for `RestoreFootprintOp`. Functionally, the same automatic
+restoration behavior that exists in `InvokeHostFunctionOp` could be replicated in `ExtendFootprintTTLOp` to provide a single
+"admin" op type for managing lifetimes and restoration.
+
+Currently, `ExtendFootprintTTLOp` is rarely called, as most rent management is performed via contract calls. With the addition of
+automatic restoration, I expect that `RestoreFootprintOp` will also be rarely invoked. It may make sense from a maintenance and UX
+perspective to deprecate `RestoreFootprintOp` in favor of a single op type.
+
+#### getledgerentry Endpoint
+
+In order to facilitate state queries for RPC preflight, captive-core will expose the following HTTP endpoint:
+
+`getledgerentry`
+
+Used to query both live and archived LedgerEntries. While `getledgerentryraw` does simple key-value lookup
+on the live ledger, `getledgerentry` will query a given key in both the live BucketList and Hot Archive BucketList.
+It will also report whether an entry is archived, evicted, or live, and return the entry's current TTL value.
+
+A POST request with the following body:
+
+```http
+ledgerSeq=NUM&key=Base64&key=Base64...
+```
+
+- `ledgerSeq`: An optional parameter, specifying the ledger snapshot to base the query on.
+  If the specified ledger in not available, a 404 error will be returned. If this parameter
```

Comment:
it seems that a 404 error can be ambiguous because it can be caused by the ledger not being available or by the key not being present.